packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -0,0 +1,130 @@
+import retrying
+
+LATENCY = 60
+
+
+@retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
+def test_metrics_ping(cluster):
+    """ Test that the metrics service is up.
+    """
+    for agent in cluster.slaves:
+        response = cluster.metrics.get('ping', node=agent)
+        assert response.status_code == 200, 'Got non-200 status code '
+        'from private agent.'
+        assert response.json()['ok'], 'Metrics /ping is not ok on private '
+        'agent.'
+
+    for agent in cluster.public_slaves:
+        response = cluster.metrics.get('ping', node=agent)
+        assert response.status_code == 200, 'Got non-200 status code '
+        'from public agent.'
+        assert response.json()['ok'], 'Metrics /ping is not ok on public agent.'
+
+    for master in cluster.masters:
+        response = cluster.metrics.get('ping', node=master)
+        assert response.status_code == 200, 'Got non-200 status code '
+        'from public agent.'
+        assert response.json()['ok'], 'Metrics /ping is not ok on master.'
+
+
+@retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
+def test_metrics_node(cluster):
+    """Test /system/metrics/api/v0/node endpoint returns
+    correct metrics.
+    """
+    expected_node_metric_datapoints = [
+        'uptime',
+        'processes',
+        'cpu.cores',
+        'load.1min',
+        'load.5min',
+        'load.15min',
+        'cpu.total',
+        'cpu.user',
+        'cpu.system',
+        'cpu.idle',
+        'cpu.wait',
+        'memory.total',
+        'memory.free',
+        'memory.buffers',
+        'memory.cached',
+        'swap.total',
+        'swam.free',
+        'swap.used']
+
+    def expected_datapoint_response(response):
+        for datapoint in response["datapoints"]:
+            if datapoint["name"] not in expected_node_metric_datapoints:
+                return False
```

Comment:
> that way we can be sure the metrics we get in the response are the ones we intend to set.
That was my point :smile:

> If we only check the length, the length could be anything
I didn't mean to suggest to only check the length, but to check the length in addition to the mandatory / required datapoints in the response. You could check the length to ensure that no _new_ datapoints have been added that you weren't expecting to see in this test.